client/src/pages/spectrogram/hooks/__tests__/use-get-image.test.tsx

```diff
@@ -0,0 +1,152 @@
+import { test, describe } from 'vitest';
+import { useGetImage } from '../use-get-image';
+import { renderHook, waitFor } from '@testing-library/react';
+import { TILE_SIZE_IN_IQ_SAMPLES, COLORMAP_DEFAULT } from '@/utils/constants';
+import { colMaps } from '@/utils/colormap';
+import { Simulate } from 'react-dom/test-utils';
+
+// 16 elements exactly
+const BucketsOfDb = [-100, -90, -80, -70, -60, -50, -40, -30, -20, -10, 0, 10, 20, 30, 40, 50];
+
+// types of sample recordings
+enum SampleType {
+  MultipleBuckets = 0,
+  WhiteBox = 1,
+}
+
+const normalizeMagnitude = (magnitude_in_db: number, magnitudeMin: number, magnitudeMax: number) => {
+  const dbPer1 = 255 / (magnitudeMax - magnitudeMin);
+  let magnitude = (magnitude_in_db - magnitudeMin) * dbPer1; // normalize to 0-255
+  magnitude = magnitude > 255 ? 255 : magnitude; // clip above 255
+  magnitude = magnitude < 0 ? 0 : magnitude; // clip below 0
+  return magnitude;
+};
+
+const generateSampleRecording = (
```

Comment:
I would move this to a function within the utilities folder

Reply:
done